core/model/model.go

```diff
@@ -123,3 +124,45 @@ func (u UUID) Validate() error {
 	}
 	return nil
 }
+
+// Qualifier represents a string type used
+// to disambiguate a model name.
+type Qualifier string
+
+// String implements [Stringer].
+func (q Qualifier) String() string {
+	return string(q)
+}
+
+var (
+	validModelNameSnippet = regexp.MustCompile(`^[a-z0-9]+[a-z0-9-]*$`)
+)
+
+// Validate returns an error if the model qualifier is not valid.
+func (q Qualifier) Validate() error {
+	if !validModelNameSnippet.MatchString(q.String()) || len(q.String()) > 63 {
+		return errors.Errorf("model qualifier %q %w", q, coreerrors.NotValid)
+	}
+	return nil
+}
+
+// QualifierFromUserTag returns a model qualifier created
+// from the supplied user tag.
+func QualifierFromUserTag(u names.UserTag) Qualifier {
+	validQualifier := strings.ToLower(u.Id())
+	// Replace chars from a valid user tag that we
+	// don't want in a qualifier with "-".
+	validQualifier = strings.NewReplacer(
+		".", "-", "+", "-", "@", "-",
+	).Replace(validQualifier)
+	return Qualifier(validQualifier)
+}
+
+// UserTagFromQualifier creates a valid user tag
+// from the supplied qualifier.
+func UserTagFromQualifier(q Qualifier) (names.UserTag, error) {
+	if err := q.Validate(); err != nil {
+		return names.UserTag{}, err
+	}
+	return names.NewUserTag(q.String()), nil
+}
```

Comment:
Is a qualifier really transmutable to a user tag? I think the name for this and the documentation needs to change.

Something like `ApproximateUserTagFromQualifier` and a comment that this function should only be used for legacy facades/legacy offer urls.